### PR TITLE
Support subdomain forwarding for external A records

### PR DESCRIPTION
**Pull request by paulopmt1**
Related to 4347-gh-Automattic/dotcom-forge

## Proposed Changes

* Update 1: Allow users to edit Subdomains Forwardings if the NS is pointing to us and the DNS A record is pointing away:

	| Before | After |
	|--------|--------|
	|image | image | 

* Update 2: Allow users to edit Domain Forwarding even if it's disabled (to match "DNS records" strategy)

	| Before | After |
	|--------|--------|
	| image | image | 

* Update 3: Show a warning message if NS servers are not pointing to DOTCOM: 

	| Before | After |
	|--------|--------|
	|image | image | 

	The message is similar to the DNS records:

	image

## Testing Instructions

**Update 1**
* Using a valid domain, change the DNS A record to point outside wordpress.com
* Ensure the warning "You can only forward subdomains. To forward..." will show when DNS A is pointing away (it can take a bit to start showing)
* Apply this diff: D126993-code
* Ensure you can only add subdomain forwarding
* Create a new subdomain forwarding and test if it works as expected

**Update 2 and 3**
* Configure your NS server to be outside wpcom
* Ensure you see the warning message
* Ensure you still can edit Domain Forwarding
* Ensure you won't see two warnings ad the same time. If the NS server is pointing away, we don't expect to show the warning "You can only forward subdomains. To forward..."

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by DustyReagan — APPROVED** [review]
Tests well!

**Comment by paulopmt1** [comment]
Waiting for the backend approval (D126993-code) to merge it

**Pull request merged**
